### PR TITLE
E2e test for TF serving with GPU

### LINKED ISSUE #291: E2E Test for TFServing with GPUs
We need an E2E test for TF Serving with GPUs.

As part of this we should built it continuously with prow.

### PR DESCRIPTION
**Pull request by lluunn**
Fix #291 
- refactor workflow to avoid code duplication
- Append cpu/gpu to the workflow name and junit name so that they can be distinguished
- Added result file for GPU. It's slightly different (0.0001) than CPU result (not sure why)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/442)

### CODE REVIEW

**Comment by lluunn** [comment]
presubmit shows succeeded, but the test failed...

**Comment by lluunn** [comment]
/cc @jlewi 
PTAL

File: components/k8s-model-server/images/releaser/components/workflows.libsonnet [hunk 1]
```diff
@@ -313,40 +372,30 @@
                 mirrorVolumeMounts: true,
               }],
             ),  // deploy-tf-serving
+            buildTemplate(
+              "deploy-tf-serving-gpu",
+              deploy_tf_serving_gpu_command,
+              [
+                {
```
Comment (jlewi) [inline components/k8s-model-server/images/releaser/components/workflows.libsonnet]: Are these environment variables correct? Why do you need to use docker in docker when you deploy TF serving?

File: testing/test_deploy.py [hunk 2]
```diff
@@ -224,7 +223,7 @@ def teardown(args):
   core_api.delete_namespace(args.namespace, {})
 
 def determine_test_name(args):
-  return args.func.__name__
+  return args.func.__name__ + "-" + args.deploy_name if args.deploy_name else args.func.__name__
```
Comment (jlewi) [inline testing/test_deploy.py]: Make this a multi-line if/else statement; its easier to read then squashing it on a single line.

File: components/k8s-model-server/images/releaser/components/workflows.libsonnet (hunk 1 above)
Comment (lluunn) [reply components/k8s-model-server/images/releaser/components/workflows.libsonnet]: We needed because we were building the image. Removing now.

**Comment by lluunn** [comment]
updated the PR description, and squashed.
PTAL, thanks

**Pull request merged**